internal/codegen/tpl_stencil_arg.go

```diff
@@ -205,17 +205,8 @@ func (s *TplStencil) validateArg(pth string, arg *configuration.Argument, v inte
 	if err := schema.Validate(v); err != nil {
 		var validationError *jsonschema.ValidationError
 		if errors.As(err, &validationError) {
-			// If there's only one error, return it directly, otherwise
-			// return the full list of errors.
-			errs := validationError.DetailedOutput().Errors
-			out := ""
-			if len(errs) == 1 {
-				out = errs[0].Error
-			} else {
-				out = fmt.Sprintf("%#v", validationError.DetailedOutput().Errors)
-			}
-
-			return fmt.Errorf("module %q argument %q validation failed: %s", s.t.Module.Name, pth, out)
```

Comment:
I really like the spirit of this change. I think having the full path is very helpful, but I am worried that this will be too much information and overwhelm the user.

Instead of using the `pth` here, could we make a nice output using a combination of `AbsoluteKeywordLocation` and `InstanceLocation`?

Reply:
Cool, yeah this PR captures the spirit without any polish.

There is a tricky detail here which is that `DetailedOutput` is recursive and I hadn't quite decided how to handle that, but I suppose we can just provide some recursive renderer that just captures the two fields you mentioned.  What do you think of that approach?

Reply:
I think there's a better way for us to represent this for sure. I'd also like to print out this information instead of returning it in the error signature, that would help us represent this better.

I'm open to giving you an idea of how to do it, or I can take it on myself here real quick. Let me know! 

Thanks so much for the contribution and conversation starter though :heart:

Reply:
Please feel free to take it, I’m happy to just be the conversation starter here. Thanks Artem and Jared.